### ISSUE TITLE
Specify log level as a parameter

### ISSUE THREAD

**Issue by daroczig** [zatonovo/futile.logger#25]
First of all, thank you very much for this extremely useful package!

I am using it in several packages to auto-generate logs on what data transformations were done at my projects. Things are getting complicated as the number of functions grew, and it turned out that it might not be the best idea to `flog.info` the number of rows fetched from a database when such query occurs a dozen times in another function, where I'd like to focus on some other items to be logged as `INFO`, and keep the previous `INFO` messages as rather `DEBUG`. This is a simple example, but this can get complicated, believe me :)

In short, I'd like to check if I'm logging in nested or top-level function, and return messages of nested functions with a lower priority. Unfortunately it's not (easily) possible via the `flog.info`, `flog.debug` etc. functions -- can we have a `flog` function that takes the logs with the level specified as an argument?

**Comment by muxspace** [comment]
Hi, thanks for the kind words.

There's a little known feature that might be of use to you. Essentially you can create multiple loggers each with their own hierarchical namespace. You can control the logging threshold for each namespace.
```R
> flog.threshold(WARN, name='inner')
NULL
> flog.info("foo")
INFO [2015-04-01 10:13:09] foo
> flog.info("boo", name='inner')
```
The 'ROOT' logger is still set at INFO, while the 'inner' logger is set to WARN. The logging threshold is inherited for any other loggers that you create within 'inner':
```R
> flog.info("woo", name='inner.b')
```
So 'inner.b' also has the log threshold set to WARN. Of course, you can set it to whatever you want, but by default it inherits from the parent namespace.

In your situation, I suggest using a different namespace for your database calls like 'db'. By setting the threshold to WARN, they will be turned off. Then if you need to see the messages, just change the threshold back to INFO. Suppose your result set is in a data.frame named `df`. You can do something like this:
```R
# Turn off INFO messages
flog.threshold(WARN, name='db')
# In your code, do this
flog.info("Row count: %s", nrow(df))

# Later when you want to see the messages, change the threshold back to INFO
flog.threshold(INFO, name='db')
```

**Comment by daroczig** [comment]
Thank you very much.

After all, I think I will use `futile.logger:::.log_level`, as there would be way too many different namespaces. E.g. I have helper functions loading data from databases (with info, debug and trace messages), then a wrapper which loads similar information from different databases and merges that, which can be also wrapped inside of an aggregation function, which is called from a reporting fn etc.

**Comment by muxspace** [comment]
Yeah, without seeing the full picture it's hard to say whether one approach is better than another. That said, do keep in mind that private package functions can change without notice, since they are not meant to publicly consumed.
